flink-connector-examples/doc/watermark/README.md

```diff
@@ -0,0 +1,96 @@
+
+### Usage of Pravega source with watermark 
+There is a slight difference that Flink forces that event timestamp can be extracted from each record, while Pravega as a streaming storage doesn't have that limitation.
+
+In order to enable Pravega watermark to transfer into Flink, Flink readers accepts an implementation of
+1. How to extract timestamp from each record
+2. How to leverage the watermark timestamp given the time bound from Pravega readers.
+
+This thought is abstracted into an interface called `AssignerWithTimeWindows`. It's up to you to implement it.
+While building the reader, please use `withTimestampAssigner(new MyAssignerWithTimeWindows())` to register the assigner. 
+
+### Usage of Pravega sink with watermark 
+The application reads text from a socket, assigns the event time and then propagating the Flink watermark into Pravega with `enableWatermark(true)` in the Flink writer.
+
+### Our Recursive Application
+`EventTimeAverage` reads sensor data from the stream with Pravega watermark, calculates an average value for each sensor under an fixed-length event-time window and generates the summary sensor data back into another Pravega stream.
+You can run it recursively by reusing the result of the smaller window.
+
+## How to run the application and verify
+The scripts can be found under the flink-examples directory in:
+```
+flink-connector-examples/build/install/pravega-flink-examples/bin
+```
+
+Start the `PravegaWatermarkIngestion` app in one window by
+```
+$ bin/pravegaWatermarkIngestion [-controller tcp://localhost:9090]
+```
+
+and then start several `EventTimeAverage` applications in parallel
+
+Required parameters:
+1. `input` for input stream Name, default is `raw-data` which contains data from `PravegaWatermarkIngestion`
+2. `output` for output stream Name
+3. `window` for event time window length, set it increasingly in the recursive runs.
+For example,
+```
+$ bin/eventTimeAverage [-controller tcp://localhost:9090] --input raw-data --output avg-10s --window 10
+$ bin/eventTimeAverage [-controller tcp://localhost:9090] --input avg-10s --output avg-500s --window 500
+$ bin/eventTimeAverage [-controller tcp://localhost:9090] --input avg-500s --output avg-final --window 10000
+```
+
+You can see the windowing average statistics recursively output for each job with different window lengths.
+All the jobs are running under event time clock perfectly.
+
+Sample output:
+
+```
+// 10s avg
+...
+2> SensorData{sensorId=0, value=-0.17122998340916468, timestamp=1500010000000}
+2> SensorData{sensorId=1, value=0.7330542299427434, timestamp=1500010000000}
+2> SensorData{sensorId=2, value=0.9633717649380801, timestamp=1500010000000}
+```
+
+```
+// 500s avg from 10s
+...
+2> SensorData{sensorId=2, value=-0.05116250806600923, timestamp=1500009500000}
+2> SensorData{sensorId=0, value=0.12731348232574283, timestamp=1500009500000}
+2> SensorData{sensorId=1, value=0.07047071003831529, timestamp=1500009500000}
```

Comment:
If you get the chance, please adjust the `SensorData::toString` to format the timestamp to be human-readable.

Reply:
Fixed